[ Samuel Lau ] iP

Samuel-bit-prog opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotesl)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here.](https://github.com/Samuel-bit-prog/ip)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:


- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Samuel-bit-prog committed:
Level-1

Samuel-bit-prog committed:
Level-2

Samuel-bit-prog committed:
Level-3 done

Samuel-bit-prog committed:
Level-4 done

Samuel-bit-prog committed:
A-TextUiTesting done

Samuel-bit-prog committed:
Level-5 done

Samuel-bit-prog committed:
Level-6 done

Samuel-bit-prog committed:
Add filewriter that works for an example

Samuel-bit-prog committed:
Level-7 done

Samuel-bit-prog committed:
Level-8 done

Samuel-bit-prog committed:
Merge branch 'branch-Level-8'

TLChicken commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,155 @@
+        readAll();
+        Scanner s = new Scanner(System.in);
+        String command = s.nextLine();
+        while(!command.equals("bye")) {
```
Perhaps it would be better if there was a space between the word "while" and the opening parenthesis? 
Reference: Java reserved words should be followed by a white space.

TLChicken commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,155 @@
+        Scanner s = new Scanner(System.in);
+        String command = s.nextLine();
+        while(!command.equals("bye")) {
+            switch(command.split(" ")[0]) {
```
Same for this switch statement, it might look better if there was a space between the word "switch" and the opening parenthesis.
Reference: Java reserved words should be followed by a white space.

TLChicken commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,155 @@
+        }
+
+        try {
+            BufferedReader b = new BufferedReader(new FileReader(f2));
```
Maybe it would help make the code clearer if b was named in a more descriptive way? The length of code that uses it seems long enough to give it a more descriptive name. Same thing for some of the other variables used throughout the code.
Reference: Variables with a large scope should have long names, variables with a small scope can have short names.

amzhy replied:
Variable `b` seems to have a large enough scope - it could have a longer, more descriptive name? I saw a similar naming for the `Scanner` variable  `s` below as well.

TLChicken reviewed (commented):
Nice, LGTM! The code is very clean and concise and almost all of it follows the coding standards. It would be great if some JavaDocs were added though as they are part of the requirements.

amzhy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,155 @@
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
```
Looks great, but java imports should be in lexographical order and could show up as checkstyle errors later. Perhaps it could be rearranged in this order:
```
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;
```

amzhy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,155 @@
+                command = s.nextLine();
+                break;
+            case "deadline":
+                String[] splitD = command.split(" ", 2)[1].split(" /by ", 2);
```
Perhaps `splitD` could be renamed as `splitDeadline`, similar to `splitE` below?

amzhy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,155 @@
+                String[] splitE = command.split(" ", 2)[1].split(" /at ", 2);
+                String one = splitE[0];
+                String two = splitE[1];
+                Event toAdd3 = new Event(one, two);
```
Maybe it would be better to not use [numbers or case](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html) such as `toAdd3` to distinguish names?

amzhy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,155 @@
+        String logo = "DUKE\n";
+        System.out.println("Hello from\n" + logo + "What can I do for you?");
+        readAll();
+        Scanner s = new Scanner(System.in);
```
Variable `s` seems to have a large enough scope - it could have a longer, more descriptive name? I saw a similar naming for the `BufferedReader` variable  `b` above as well.

amzhy reviewed (commented):
LGTM 👍  Just a few minor details to fix :)

Samuel-bit-prog committed:
modified DukeException and more oop for storage, TaskList and ui

Samuel-bit-prog committed:
Made the code more OOP

Samuel-bit-prog committed:
Moved classes into package

Samuel-bit-prog committed:
add junit test

Samuel-bit-prog committed:
add javadocs

Samuel-bit-prog committed:
Merge pull request #2 from Samuel-bit-prog/branch-A-JavaDoc

add javadocs

Samuel-bit-prog committed:
Tweak the code to comply with a coding standard.

Samuel-bit-prog committed:
Merge branch 'master' into branch-A-CodingStandard

Samuel-bit-prog committed:
Merge pull request #3 from Samuel-bit-prog/branch-A-CodingStandard

Tweak the code to comply with a coding standard.

Samuel-bit-prog committed:
Give users a way to find a task by searching for a keyword.

Samuel-bit-prog committed:
Merge branch 'master' into branch-Level-9

Samuel-bit-prog committed:
Merge pull request #4 from Samuel-bit-prog/branch-Level-9

Give users a way to find a task by searching for a keyword.

Samuel-bit-prog committed:
Fix javadoc errors

Samuel-bit-prog committed:
Merge pull request #5 from Samuel-bit-prog/add-gradle-support

Add Gradle support

Samuel-bit-prog committed:
add checkstyle

Samuel-bit-prog committed:
Merge pull request #6 from Samuel-bit-prog/branch-A-Gradle

Branch a gradle

Samuel-bit-prog committed:
add GUI first version

Samuel-bit-prog committed:
Merge pull request #7 from Samuel-bit-prog/branch-level-10

add GUI first version.

Samuel-bit-prog committed:
Change build.gradle mainClassName

mainClassName for application in build gradle is wrong.

Having the right name is needed for the creation of a proper JAR file.

Let's edit the build.gradle to have the right mainClassName declaration.

Samuel-bit-prog committed:
Add assertions

Assertions are used to define assumptions about the program state.

It is important so that the runtime can verify these assumptions.

Let's
* assert that command is not null for Parser methods.
* assert that task.list is not null for writeAll method in Storage class.

Samuel-bit-prog committed:
Merge pull request #8 from Samuel-bit-prog/branch-A-Assertions

Add assertions

Samuel-bit-prog committed:
Improve code quality

Code quality includes run-tim efficiency, security, robustness and more.

It is important for readability and understandability.

Let's
* put default cases for all usages of switch.
* use checkstyle and fix any style errors.
* remove string concatenation in loop and use StringBuilder instead.
* add missing javadocs.

Samuel-bit-prog committed:
Merge pull request #9 from Samuel-bit-prog/branch-A-CodeQuality

Improve code quality

Samuel-bit-prog committed:
Add B-DoWithinPeriodTasks

Samuel-bit-prog committed:
Improve GUI

Samuel-bit-prog committed:
Improve README.md and add Ui.png

Samuel-bit-prog committed:
Set theme jekyll-theme-slate

Samuel-bit-prog committed:
Fix README.md

Samuel-bit-prog committed:
Merge remote-tracking branch 'origin/master'

Samuel-bit-prog committed:
Set theme jekyll-theme-minimal

Samuel-bit-prog committed:
Fix README.md error

Samuel-bit-prog committed:
Merge remote-tracking branch 'origin/master'

Samuel-bit-prog committed:
Set theme jekyll-theme-slate

Samuel-bit-prog committed:
Set theme jekyll-theme-slate

Samuel-bit-prog committed:
Set theme jekyll-theme-slate

Samuel-bit-prog committed:
Fix checkstyle error

Samuel-bit-prog committed:
Set theme jekyll-theme-cayman

Samuel-bit-prog committed:
Set theme jekyll-theme-slate

Samuel-bit-prog committed:
Set theme jekyll-theme-merlot

Samuel-bit-prog committed:
Set theme jekyll-theme-slate

Samuel-bit-prog committed:
Fix README.md

Samuel-bit-prog committed:
Fix error handling issues.
